torch_xla_py/data_parallel.py

```diff
@@ -180,11 +180,11 @@ def __init__(self, network, device_ids=None, batchdim=0, drop_last=False):
     self._batchdim = batchdim
     self._drop_last = drop_last
     self._native_run = False
-    if self._device_ids:
+    if not self._device_ids or len(self._device_ids) == 1:
```

Comment:
Is there ever a case where we have 1 device and yet we want replication? If so, I will check only for "self._device_ids is None" and return None in xla_replication_devices instead of the size 1 list

Reply:
Replication can still be `not None` with one device, but this is probably the shortest change:
```Diff
diff --git a/torch_xla_py/data_parallel.py b/torch_xla_py/data_parallel.py
index 9c06a30..581e8be 100644
--- a/torch_xla_py/data_parallel.py
+++ b/torch_xla_py/data_parallel.py
@@ -180,7 +180,7 @@ class DataParallel(object):
     self._batchdim = batchdim
     self._drop_last = drop_last
     self._native_run = False
-    if self._device_ids:
+    if len(self._device_ids) > 1:
       replication_devices = xm.xla_replication_devices(self._device_ids)
       self._replication = xm.Replication(self._device_ids, replication_devices)
     else:
```

Which also work with the `--num_cores=0` which is PyTorch/CPU.